src/aiida/engine/daemon/execmanager.py

```diff
@@ -62,9 +62,9 @@ def _find_data_node(inputs: MappingType[str, Any], uuid: str) -> Optional[Node]:
     return data_node
 
 
-def upload_calculation(
+async def upload_calculation(
     node: CalcJobNode,
-    transport: Transport,
+    transport: Union['Transport', 'AsyncTransport'],
```

Comment:
```suggestion
    transport: 'Transport | AsyncTransport',
```

Reply:
If I see a type annotation like this, it usually means defining a protocol as interface is a better solution ;) But I won't bother in this PR.